### PR TITLE
Scoring: exclude_from_ranking

### PR DESCRIPTION
**Pull request by chrisvanrun**
Part of the tiny pitch:
- https://github.com/DIAGNijmegen/rse-roadmap/issues/322

This PR adds an optional (default: false) `exclude_from_ranking ` field in the extra results columns JSON definition. Setting it to `true` allows additional metrics to be shown on Challenge leaderboards while being ignored while calculating the rank.

The related test is a construction copy of `test_calculate_ranks` since I did not find it wise to extend that test with _yet_ another factor. It would have pushed the 12 (2x3x2) to 24 which is over the top imho.

### CODE REVIEW

**Comment by jmsmkn** [comment]
I would have thought that the template and view would need updating too as the rank will not be present for excluded fields, right?

File: app/tests/evaluation_tests/test_utils.py [hunk 1]
```diff
@@ -136,6 +136,96 @@ def test_calculate_ranks(django_assert_max_num_queries):
...
+            b_included: {
+                "ranks": [5, 4, 1, 1, 1, 0],
+                "rank_scores": [5, 3.5, 2, 2, 2, 0],
+            },
+            b_excluded: only_a,
+        },
+    }
+
+    for score_method in (Phase.ABSOLUTE, Phase.MEDIAN, Phase.MEAN):
+        for b_exclusion in (True, False, None):
```
[the remark this answers is not in the document]
Comment (chrisvanrun) [reply app/tests/evaluation_tests/test_utils.py:198]: Check!

**Comment by chrisvanrun** [comment]
> I would have thought that the template and view would need updating too as the rank will not be present for excluded fields, right?

I might have completely missed the fact that the leaderboard templating changed if the scoring method changed! I'll look into this after the weekend.

**Comment by chrisvanrun** [comment]
Refactor the test(s) and have updated the leaderboard template to not decorate the excluded metrics:
image

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**